### ISSUE TITLE
Native M1 support

### ISSUE THREAD

**Issue by jonathanherdt** [ibmdb/python-ibmdb#739]
**Is your feature request related to a problem? Please describe.**
With ticket #712 we got emulated x86 support for M1. This issue therefor is about native M1 support.

**Describe the solution you'd like**
It would be great to be able to install and run python-ibmdb with an M1 version of Python.

**Describe alternatives you've considered**
Installing Python in x86 compatibility mode. This has several drawbacks, one of them being that if you are in a large python project, all other packages will have to be x86, too, which reduces the performance.

**Additional context**
The original description of #712 still is valid for this one, of which this message during installation is most important:
```
# First download the source code from this repo's releases: https://github.com/ibmdb/python-ibmdb/releases
rm -rf clidriver && rm -rf build && python setup.py install
[...lots of messages]
ld: warning: ignoring file clidriver/lib/libdb2.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
[...lots of messages]
```

**Comment by bimalkjha** [comment]
@jonathanherdt As you know the python-ibm_db uses IBM ODBC/CLI Driver underneath for connectivity. The error "ld: warning: ignoring file clidriver/lib/libdb2.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64" tells that the libdb2.dylib is built for macOS-x86_64. This library is built by IBM and it is part of "IBM Data Server Driver Package" provided by IBM. There is not arm64 version of lib library available from IBM and hence python-ibm_db can not work with M1 version of Python.
If want to put request, please "Add a new Idea" to provide dsdriver installer for M1 Chip MacOS system here: https://ibm-data-and-ai.ideas.ibm.com/?project=DB2CON
Based on customers vote, IBM team will pick up the request and implement.
Thanks and Regards,
Bimal Jha

**Comment by jonathanherdt** [comment]
Thank you for the input. I have created a new ticket in their system under the ID DB2CON-I-92. It can be viewed here: https://ibm-data-and-ai.ideas.ibm.com/ideas/DB2CON-I-92 It is not visible yet as I guess IBM is verifying it.

**Comment by mmdanziger** [comment]
Voting to re-open. This is a serious issue that is **not** solved. Granted it is dependent on upstream components but if you close it you'll just get new issues like #741 popping up.

**Comment by bimalkjha** [comment]
@mmdanziger Please let us know why you see this issue is not addressed? If you are facing any issue, please provide the details.
The pre-requirement for ibm_db to work on M1 Chip system is to have gcc installation as MacOS12.x is not coming with gcc. Many users are already using ibm_db on M1 system and it is working fine for us too. So, there is no issue with support. We are always open to look into new issues. Thanks.

**Comment by bimalkjha** [comment]
@jonathanloske @mmdanziger We have released `ibm_db v3.2.5` with native installation support for MacOS Silicon Chip system and no need to use the emulated environment with x86_64 python on M1 chip system. Please install latest version of ibm_db and verify. Thanks.
